core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -195,18 +192,107 @@ private static void callAuthorizationPluginImpl(
   }
 
   public static boolean needApplyAuthorizationPluginAllCatalogs(SecurableObject securableObject) {
-    // TODO: Add `supportsSecurableObjects` method for every privilege to simplify this code
     if (securableObject.type() == MetadataObject.Type.METALAKE) {
       List<Privilege> privileges = securableObject.privileges();
       for (Privilege privilege : privileges) {
-        if (!pluginNotSupportsPrivileges.contains(privilege.name())) {
+        if (privilege.supportsMetadataObjectType(MetadataObject.Type.CATALOG)) {
           return true;
         }
       }
     }
     return false;
   }
 
+  // Check every securable object whether exists and is imported.
+  public static void checkSecurableObject(String metalake, MetadataObject object) {
+    NameIdentifier identifier = MetadataObjectUtil.toEntityIdent(metalake, object);
+
+    Supplier<NoSuchMetadataObjectException> exceptionToThrowSupplier =
+        () ->
+            new NoSuchMetadataObjectException(
+                "Securable object %s doesn't exist", object.fullName());
+
+    switch (object.type()) {
+      case METALAKE:
+        check(
+            GravitinoEnv.getInstance().metalakeDispatcher().metalakeExists(identifier),
+            exceptionToThrowSupplier);
+        break;
+
+      case CATALOG:
+        check(
+            GravitinoEnv.getInstance().catalogDispatcher().catalogExists(identifier),
+            exceptionToThrowSupplier);
+        break;
+
+      case SCHEMA:
+        check(
+            GravitinoEnv.getInstance().schemaDispatcher().schemaExists(identifier),
+            exceptionToThrowSupplier);
+        break;
+
+      case FILESET:
+        check(
+            GravitinoEnv.getInstance().filesetDispatcher().filesetExists(identifier),
+            exceptionToThrowSupplier);
+        break;
+
+      case TABLE:
+        check(
+            GravitinoEnv.getInstance().tableDispatcher().tableExists(identifier),
+            exceptionToThrowSupplier);
+        break;
+
+      case TOPIC:
+        check(
+            GravitinoEnv.getInstance().topicDispatcher().topicExists(identifier),
+            exceptionToThrowSupplier);
+        break;
+
+      default:
+        throw new IllegalArgumentException(
+            String.format("Doesn't support the type %s", object.type()));
+    }
+  }
+
+  public static void checkPrivilege(
```

Comment:
I think we can combine this method and above together, so we don't have to access the DB twice.